lib/discourse_chat/provider/webex/webex_provider.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module DiscourseChat::Provider::WebexProvider
+  PROVIDER_NAME = "webex".freeze
+  PROVIDER_ENABLED_SETTING = :chat_integration_webex_enabled
+  CHANNEL_PARAMETERS = [
+    { key: "name", regex: '^\S+$', unique: true },
+    { key: "webhook_url", regex: '^https:\/\/webexapis\.com\/v1\/webhooks\/incoming\/[A-Za-z0-9\-@\/]+\S+$', unique: true, hidden: true }
+  ]
+
+  def self.trigger_notification(post, channel, rule)
+    message = get_message(post)
+    uri = URI(channel.data['webhook_url'])
+
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = (uri.scheme == 'https')
+
+    req = Net::HTTP::Post.new(uri, 'Content-Type' => 'application/json')
+    req.body = message.to_json
+    response = http.request(req)
+
+    unless response.kind_of? Net::HTTPSuccess
+      if response.body.include?('Invalid webhook URL')
+        error_key = 'chat_integration.provider.webex.errors.invalid_channel'
+      else
+        error_key = nil
+      end
+      raise ::DiscourseChat::ProviderError.new info: { error_key: error_key, request: req.body, response_code: response.code, response_body: response.body }
+    end
+
+  end
+
+  def self.get_message(post)
+    display_name = "@#{post.user.username}"
+    full_name = post.user.name || ""
+
+    topic = post.topic
+
+    category = ''
+    if topic.category&.uncategorized?
+      category = "[#{I18n.t('uncategorized_category_name')}]"
+    elsif topic.category
+      category = (topic.category.parent_category) ? "[#{topic.category.parent_category.name}/#{topic.category.name}]" : "[#{topic.category.name}]"
+    end
+
+    message = {
+      "markdown": "**#{topic.title}**: #{category} #{topic.tags.present? ? topic.tags.map(&:name).join(', ') : ''}(#{post.full_url}) from #{full_name}(#{display_name}): #{post.excerpt(SiteSetting.chat_integration_webex_excerpt_length, text_entities: true, strip_links: true, remap_emoji: true)}"
+    }
+
+    message
```

Comment:
This line is a little long. How about something like
```suggestion
    markdown = "**#{topic.title}**: #{category} "
    markdown += "#{topic.tags.map(&:name).join(', ')} " if topic.tags.present?
    markdown += "(#{post.full_url}) from #{full_name} (#{display_name}) "
    markdown += post.excerpt(SiteSetting.chat_integration_webex_excerpt_length, 
                             text_entities: true, 
                             strip_links: true, 
                             remap_emoji: true
                            )
    { markdown: markdown }
```

Reply:
Sure; I'm up for improving things. I just did the minimum viable so far, to limit how much work time I'd spend on it :)

I'll add the tests and fix the long strings shortly.

Reply:
Do you have guidelines for line length? Should I also adjust the others, and if so; how do you feel about operators -- end of line or beginning of line, etc. For example:

```
    elsif topic.category
      category = (topic.category.parent_category)
               ? "[#{topic.category.parent_category.name}/#{topic.category.name}]"
               : "[#{topic.category.name}]"
```
or
```
      raise ::DiscourseChat::ProviderError.new info: { error_key: error_key,
                                                       request: req.body,
                                                       response_code: response.code,
                                                       response_body: response.body }
```

Is that good?

Reply:
That all looks good to me. We don't have any super strict rules on formatting or line length, so as long as it's readable, it's fine. The rules we do have are enforced by the rubocop config (which looks like it's passing 😄 )